lib/engines/hcktest/tests.rb

```diff
@@ -45,6 +45,10 @@ def list_tests(log: false)
       retry
     end
 
+    def load_tests_results
+      @tests_results = @tools.list_all_results(@target['key'], @client.name, @tag)
```

Comment:
Let's compute what `new_done_results` computes here to group the code modelling the change of test results into one method.